lib/ui/payment/payment_web_page.dart

```diff
@@ -0,0 +1,249 @@
+  }
+
+  Uri _getPaymentUrl(String paymentToken) {
+    final queryParameters = {
+      'productID': widget.planId,
+      'paymentToken': paymentToken,
+      'action': widget.actionType,
+      'redirectURL': kWebPaymentRedirectUrl,
+    };
+    var tryParse = Uri.tryParse(kWebPaymentBaseEndpoint);
+    if (kDebugMode && kWebPaymentBaseEndpoint.startsWith("http://")) {
+      return Uri.http(tryParse.authority, tryParse.path, queryParameters);
+    } else {
+      return Uri.https(tryParse.authority, tryParse.path, queryParameters);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (initPaymentUrl == null) {
+      return loadWidget;
+    }
+    return WillPopScope(
+        onWillPop: () async => _buildPageExitWidget(context),
+        child: Scaffold(
+          appBar: AppBar(
+            title: const Text('subscription'),
+          ),
+          body: Column(
+            children: <Widget>[
+              (progress != 1.0)
+                  ? LinearProgressIndicator(value: progress)
+                  : Container(),
+              Expanded(
+                child: InAppWebView(
+                  initialUrlRequest: URLRequest(url: initPaymentUrl),
+                  onProgressChanged:
+                      (InAppWebViewController controller, int progress) {
+                    setState(() {
+                      this.progress = progress / 100;
+                    });
+                  },
+                  initialOptions: InAppWebViewGroupOptions(
+                    crossPlatform: InAppWebViewOptions(
+                      useShouldOverrideUrlLoading: true,
+                    ),
+                  ),
+                  shouldOverrideUrlLoading:
+                      (controller, navigationAction) async {
+                    var loadingUri = navigationAction.request.url;
+                    _logger.info("Loading url $loadingUri");
+                    // handle the payment response
+                    if (_isPaymentActionComplete(loadingUri)) {
+                      await handlePaymentResponse(loadingUri);
+                      return NavigationActionPolicy.CANCEL;
+                    }
+                    return NavigationActionPolicy.ALLOW;
+                  },
+                  onConsoleMessage: (controller, consoleMessage) {
+                    _logger.info(consoleMessage);
+                  },
+                  onLoadStart: (controller, navigationAction) async {
+                    if (!_dialog.isShowing()) {
+                      await _dialog.show();
+                    }
+                  },
+                  onLoadError: (controller, navigationAction, code, msg) async {
+                    if (_dialog.isShowing()) {
+                      await _dialog.hide();
+                    }
+                  },
+                  onLoadHttpError: (controller, navigationAction, code, msg) async {
+                    _logger.info("onHttpError with $code and msg = $msg");
```

Comment:
Should we log the error to Sentry?

Reply:
Failure could be also due to bad internet.. Don't see the need of logging that to sentry.. 
But can still change it. We can turn if off in the future if it becomes noisy (🤞)